src/components/download-button/download-button.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import Button from '@mapbox/mr-ui/button';
+import IconText from '@mapbox/mr-ui/icon-text';
+import PropTypes from 'prop-types';
+
+export default class DownloadButton extends React.PureComponent {
+  render() {
+    const { href, text } = this.props;
+    const fileType = href.split('.').pop().toUpperCase();
+    const fileName = href.split('/').pop();
+
+    return (
+      <Button
+        href={href}
+        passthroughProps={{
+          download: fileName,
+          className: 'unprose btn btn--blue round-full py12 px24 txt-s my18'
+        }}
+      >
+        <IconText iconBefore="arrow-down">
+          {text || 'Download ' + fileType + ' file'}
+        </IconText>
+      </Button>
+    );
+  }
+}
+
+DownloadButton.propTypes = {
+  href: PropTypes.string.isRequired,
+  text: PropTypes.string
```

Comment:
One last thing we'll want to do is document these props! 

We do this by adding a comment above each prop. The comment must being with double stars. Example:

```suggestion
/** Link to the asset that will be dowloaded. */
  href: PropTypes.string.isRequired,
  text: PropTypes.string
```


During the build process the catalog site will automatically add these comments to the property table. Example:
https://mapbox.github.io/dr-ui/components/#tag

https://github.com/mapbox/dr-ui/blob/c0df94f7677261e84f2173af361247bc0055918d/src/components/tag/tag.js#L55

<img width="948" alt="2021-08-23 at 10 53 AM" src="https://user-images.githubusercontent.com/2180540/130468914-a24b8aee-1647-4018-84b9-c283ae4513fa.png">

Reply:
SO COOL